application/client/src/app/ui/views/toolbar/search/input/component.ts

```diff
@@ -159,6 +157,7 @@ export class ViewSearchInput
             this.active = new ActiveSearch(this.session.search, active);
             this.input.drop();
         }
+        this._restoreNonActiveSearch();
```

Comment:
the function is used just in one place... no sense to exclude `this.input.set().value(this.session.search.state().nonActive);` into standalone method

Reply:
I've recently read about how to write code more understandable and clean in which it is recommended to avoid writing comments and instead create functions with the function name describing what the action does. It might be obvious now what the function does, but at a later point storing it in a function might help us to understand the code easier. In my opinion, writing a function here has any downsides. But I can remove it if you still want, sure.

Reply:
yes, it makes sense; self-explanation code is always good. But in this case, you should create a method `restore` and put there all, what related to restoring
```
protected restore() {
        const active = this.session.search.state().getActive();
        if (active !== undefined) {
            this.active = new ActiveSearch(this.session.search, active);
            this.input.drop();
        }
        this.input.set().value(this.session.search.state().nonActive);
}
```
and call it from `ngAfterViewInit`. In this case all looks good. Because current solution has retore-related stuff in `ngAfterViewInit` and in method `_restoreNonActiveSearch`. That's why there are 2 optional wayt:
- put all restore related to `ngAfterViewInit` (my first suggestion)
- exclude from `ngAfterViewInit` all restore related stuff to own method and call it from `ngAfterViewInit`